src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -86,35 +86,58 @@ void LogQuery(const QueryData & data)
     ChipLogDetail(Discovery, "%s", logString.c_str());
 }
 
+/// Checks if the current interface is powered on
+/// and not local loopback.
+template <typename T>
+bool IsCurrentInterfaceUsable(T & iterator)
+{
+    if (!iterator.IsUp() || !iterator.SupportsMulticast())
+    {
+        return false; // not a usable interface
+    }
+    char name[chip::Inet::InterfaceIterator::kMaxIfNameLength];
+    if (iterator.GetInterfaceName(name, sizeof(name)) != CHIP_NO_ERROR)
+    {
+        ChipLogError(Discovery, "Failed to get interface name.");
+        return false;
+    }
+
+    if (strncmp(name, "lo", 2) == 0)
```

Comment:
Not a new issue, but doesn't seem right that if I rename my interface to "lollipop" it will be skipped.

Probably the name is the wrong thing to look at altogether, and we should just inspect the address.

Reply:
I could find no better way.

For devices this may not be an issue (since they control their wifi name and it is likely LWIP anyway). Would people rename interfaces on android/iOS?

Added a TODO to auto-create an issue.

Reply:
Can't you just get the address and see if it's ::1 or inside 127.0.0.0/8 ?

Reply:
InterfaceIterator does not provide the address only InterfaceAddressIterator does. 

Absolutely possible to use AddressIterator instead and do some dedup, but I find it more bothersome than what I would be generally add as a 'fix if you are already touching this'.

Reply:
Real fix would probably to fix the interface iterator to be able to return an address iterator for that specified interface. Having 2 types of iterators that both iterate over all interfaces is strange.